src/cryptography/fernet.py

```diff
@@ -74,6 +74,13 @@ def decrypt(self, token, ttl=None):
         timestamp, data = Fernet._get_unverified_token_data(token)
         return self._decrypt_data(data, timestamp, ttl)
 
+    def ttl(self, token, ttl):
+        current_time = int(time.time())
+        timestamp, data = Fernet._get_unverified_token_data(token)
+        # Verify the token was not tampered with, but do not fail on expiry.
+        self._decrypt_data(data, timestamp, None)
```

Comment:
This will, in addition to verifying the HMAC, decrypt the data. While you're throwing that away we should probably just refactor `_decrypt_data` so that these lines (https://github.com/pyca/cryptography/blob/master/src/cryptography/fernet.py#L105-L110) are in a separate method that we can call.

Reply:
I factored it out to `def _verify_signature()`.